src/main/java/com/google/devtools/build/lib/remote/S3ActionCache2.java

```diff
@@ -200,19 +193,30 @@ public void put(String key, byte[] blob) {
         putObject(key, new PutObjectRequest(bucketName, key, new ByteArrayInputStream(blob), new ObjectMetadata()));
     }
 
+    private synchronized String getBucketOwner() {
```

Comment:
Does is make sense to cache this for the lifetime of the worker? Otherwise you'll be doing two round-trips to S3 for each put, right?